### ISSUE TITLE
CRHM_GUI fix version for Issue_432: running crash and bug

### ISSUE THREAD

**Issue by loganxingfang** [srlabUsask/crhmcode#452]
I used the fix version for Issue 432[https://github.com/srlabUsask/crhmcode/issues/432](url) and ran a few prj files. I had some crashes and bugs to report.

1. I ran a Marmot Creek project file, the .prj and obs files are here
[Marmot.zip](https://github.com/srlabUsask/crhmcode/files/15491257/Marmot.zip)

When I tried to open the prj file, the CRHM_GUI crashed.

2. I ran a researcher Di's project files, 
the prj files are here
[20072008_test_prj_files.zip](https://github.com/srlabUsask/crhmcode/files/15491278/20072008_test_prj_files.zip)

the obs files are here
[20072008_test_obs_files.zip](https://github.com/srlabUsask/crhmcode/files/15491279/20072008_test_obs_files.zip)

There two simulation runs:

run 1: S4_group150_generate_init20080101_REW_CRHM134_LFedit1.prj from 2007 1 1 to 2008 1 1, to generate int file used by run 2

run 2: S4_group150_used20080101_init20090101_REW_CRHM134_LFedit1.prj from 2008 1 1 to 2009 1 1, to use int file generated from run 1 and also to generate next int file

These runs finished and were able to generate output and int files. The only issue is the datetime stamp on x-axis in the graphic interface, these look like MS-data number not proper human readable datetime,
see attached screen shots for these two runs
![CRHM_GUI_20072008_run_LF_PM](https://github.com/srlabUsask/crhmcode/assets/28319889/2a3b7246-9ed3-4f27-a9b8-24d5b4e2b4d7)
![CRHM_GUI_20082009_run_LF_PM](https://github.com/srlabUsask/crhmcode/assets/28319889/684e30a4-af81-4aa1-8ff0-963ba465596f)

3.  I ran a researcher Di's project files, this is a longer time simulation, 5 years: S4_group150_resetparam_generate_init20120101_test_LFedit1.prj for 2007 to 2012

the prj file is here
[S4_group150_resetparam_generate_init20120101_test_LFedit1.zip](https://github.com/srlabUsask/crhmcode/files/15491330/S4_group150_resetparam_generate_init20120101_test_LFedit1.zip)

the obs files are here
[OBS_2007_20190101_RH.zip](https://github.com/srlabUsask/crhmcode/files/15491360/OBS_2007_20190101_RH.zip)
[OBS_2007_20190101_temp_wind.zip](https://github.com/srlabUsask/crhmcode/files/15491362/OBS_2007_20190101_temp_wind.zip)
[OBS_2007_20190101_prec_Srad.zip](https://github.com/srlabUsask/crhmcode/files/15491365/OBS_2007_20190101_prec_Srad.zip)

CRHM_GUI crashed at 2011-10-24T18:00. I attached the output file 
[CRHM_20072012_crash_output_1.zip](https://github.com/srlabUsask/crhmcode/files/15491398/CRHM_20072012_crash_output_1.zip)

Not sure why it crashed. Too many forcing variables from obs files or the large value of variables WS_flow@EU(1)	WS_flow_s@EU(1) in output file:
2011-10-24T18:00	3.051169E+258	2.825157E+254

**Comment by loganxingfang** [comment]
@liudilena

**Comment by jhs507** [comment]
Have you tried these in the latest release version? Are these errors unique to the version I gave to you as the fix for issue 432?

**Comment by loganxingfang** [comment]
Yes, I opened release version CRHM_GUI 1.3.4. Marmot Creek prj crashed.

Other prj files have Group ending AZ, BZ, CZ, DZ, so CRHM_GUI 1.3.4 cannot handle these. The fix version 442 and this 432 handles them.

**Comment by loganxingfang** [comment]
For the datetime display issue, it is only shown for these group prj files. Maybe something to do with large arrays of obs data: 150 arrays in each variable, and there are five variables used by these group prj files.

I ran other prj file, the datetime is displayed in the graphic area.

**Comment by loganxingfang** [comment]
> Yes, I opened release version CRHM_GUI 1.3.4. Marmot Creek prj crashed.
> 
> Other prj files have Group ending AZ, BZ, CZ, DZ, so CRHM_GUI 1.3.4 cannot handle these. The fix version 442 and this 432 handles them.

@jhs507 to verify the datetime display issue, I used CRHM_GUI 1.3.4 and ran one of the large number of group project I included in this issue: S4_group150_generate_init20080101_REW_CRHM134_LFedit1.prj

The datetime display issue is also presented in CRHM_GUI 1.3.4 regardless of its issue for Group ending AZ, BZ, CZ, DZ.

See the attached screen shot
![CRHM_GUI_20072008_run_LF_usingCRHM_GUI1 3 4](https://github.com/srlabUsask/crhmcode/assets/28319889/80c35a0a-785b-4e10-9c85-e5d3b0c7ef76)
I think there might be some exception in CRHM_GUI when handling large number group prj file reading obs files with large arrays.

**Comment by loganxingfang** [comment]
@jhs507 I also did comparison between CRHM_output files generated by this version of CRHM_GUI: CRHM_20072008_output_1_PM.obs vs CRHM_20072012_crash_output_1.obs

CRHM_20072008_output_1_PM.obs is generated from S4_group150_generate_init20080101_REW_CRHM134_LFedit1.prj file

CRHM_20072012_crash_output_1.obs is generated from S4_group150_resetparam_generate_init20120101_test_LFedit1.prj file

As the outputs are quite different after 2007 1 2, see attached screen shot
![CRHM_GUI_output_comparison](https://github.com/srlabUsask/crhmcode/assets/28319889/2e476ab9-9604-4a96-966c-6418a5284cb7)

Then, I went through the model prj and forcing obs files and found that the longer period 5-year project S4_group150_resetparam_generate_init20120101_test_LFedit1.prj, the forcing obs data file "S4_Srad_20072018_3hc.obs" does not have data before 2012 1 1; that is, there is no shortwave radiation data between 2007 1 1 and 2012 12 31. This could partially explain why the model prj crashed. See attached
![S4_Srad_20072018_3hc_obs file screen shot jpg](https://github.com/srlabUsask/crhmcode/assets/28319889/56efbae9-9472-4362-a4ed-6f005f1a81fc)
I have asked the researcher to do some quality check for the obs data files. I will report it when I find out more.

**Comment by loganxingfang** [comment]
@jhs507 I have some updates. After discussion and working with researcher on the model prj and obs files, CRHM_GUI fix version for Issue_432 can finish the continuous run from 2007 1 1 to 2018 12 31. I have the following updated prj and obs files:

updated prj file
[S4_group150_resetparam_NSoildepth10_init2019_LFedit.zip](https://github.com/srlabUsask/crhmcode/files/15506138/S4_group150_resetparam_NSoildepth10_init2019_LFedit.zip)
updated two obs files
[adjustedobstest_May302024.zip](https://github.com/srlabUsask/crhmcode/files/15506141/adjustedobstest_May302024.zip)
The model can save output file and int file:
[Model_output_and_int_files.zip](https://github.com/srlabUsask/crhmcode/files/15506146/Model_output_and_int_files.zip)
Screen shot for the model run at the finish looks fine, except for the datetime display issue discussed above. Maybe this is to do with large group prj file reading large array obs files.
![S4_group150_2007_2019_run](https://github.com/srlabUsask/crhmcode/assets/28319889/ab7476e8-0fd4-419b-9a69-b75873937b44)

As for the Marmot Creek prj run crash, perhaps it is another exception.

Overall, this version is generally improved.

**Comment by jhs507** [comment]
Glad to hear it is improved.

I have added this and a few other issues to the Bug tracker can you assign a priority to it so that I can prioritize which bugs to investigate next?
